Add hammer azure, rh-cloud and remove csv to hammer-devel

chris1984 opened the pull request:
Test failure is unrelated. Here is the playbook run, showing it worked

```
PLAY RECAP *********************************************************************
centos7-hammer-devel       : ok=30   changed=24   unreachable=0    failed=0    skipped=2    rescued=0    ignored=3
```

ekohl commented on roles/hammer_devel/tasks/hammer_config.yml:
```diff
@@ -57,11 +57,17 @@
 
-- name: 'Configure hammer-cli-csv'
-  command: cp ~/hammer-cli-csv/config/csv.yml ~/.hammer/cli.modules.d/
+- name: 'Configure hammer-cli-azure'
```
Isn't there some way to iterate in a loop here?

chris1984 replied:
I can try to refactor it so it looks cleaner with all the tasks

chris1984 commented:
@ekohl I refactored the modules, let me know what you think:

```
TASK [hammer_devel : Configure Hammer CLI Foreman and modules] *****************
changed: [centos72-hammer-devel] => (item={'key': 'hammer-cli-foreman', 'value': 'foreman.yml'})
changed: [centos72-hammer-devel] => (item={'key': 'hammer-cli-foreman-admin', 'value': 'foreman_admin.yml'})
changed: [centos72-hammer-devel] => (item={'key': 'hammer-cli-katello', 'value': 'katello.yml'})
changed: [centos72-hammer-devel] => (item={'key': 'hammer-cli-foreman-rh-cloud', 'value': 'foreman_rh_cloud.yml'})
changed: [centos72-hammer-devel] => (item={'key': 'hammer_cli_foreman_remote_execution', 'value': 'foreman_remote_execution.yml'})
changed: [centos72-hammer-devel] => (item={'key': 'hammer-cli-foreman-virt-who-configure', 'value': 'foreman_virt_who_configure.yml'})
changed: [centos72-hammer-devel] => (item={'key': 'hammer_cli_foreman_azure_rm', 'value': 'foreman_azure_rm.yml'})
Thursday 05 October 2023  14:51:10 -0400 (0:00:05.580)       0:00:22.419 ******
Thursday 05 October 2023  14:51:10 -0400 (0:00:05.580)       0:00:22.419 ******
 [started TASK: hammer_devel : Configure hammer-cli-foreman-admin logging core on centos72-hammer-devel]
```

```
PLAY RECAP *********************************************************************
centos72-hammer-devel      : ok=25   changed=8    unreachable=0    failed=0    skipped=1    rescued=0    ignored=3

Thursday 05 October 2023  14:51:15 -0400 (0:00:00.661)       0:00:27.177 ******
===============================================================================
hammer_devel : Clone the hammer repositories ---------------------------- 6.43s
hammer_devel : Configure Hammer CLI Foreman and modules ----------------- 5.58s
Gathering Facts --------------------------------------------------------- 1.74s
git : Install git ------------------------------------------------------- 1.26s
hammer_devel : Install gems --------------------------------------------- 1.09s
ruby_scl : Enable and configure SCL at login ---------------------------- 0.84s
ruby_scl : Install SCL repository --------------------------------------- 0.83s
ruby_scl : Install Ruby SCL --------------------------------------------- 0.81s
hammer_devel : Install gem native dependencies -------------------------- 0.80s
hammer_devel : Update hammer-cli-foreman host --------------------------- 0.77s
hammer_devel : Add hammer config directory ------------------------------ 0.76s
hammer_devel : Configure hammer ----------------------------------------- 0.75s
hammer_devel : Add hammer module config directory ----------------------- 0.68s
hammer_devel : Add local gems to Gemfile.local -------------------------- 0.68s
hammer_devel : Configure hammer-cli-foreman-admin logging core ---------- 0.67s
hammer_devel : Update hammer-cli-foreman username ----------------------- 0.67s
hammer_devel : Update hammer-cli-foreman password ----------------------- 0.67s
hammer_devel : Alias rake ----------------------------------------------- 0.66s
hammer_devel : Configure hammer-cli-foreman-admin logging for katello --- 0.66s
hammer_devel : Alias hammer --------------------------------------------- 0.65s
Thursday 05 October 2023  14:51:15 -0400 (0:00:00.661)       0:00:27.176 ******
===============================================================================
hammer_devel ----------------------------------------------------------- 21.65s
ruby_scl ---------------------------------------------------------------- 2.47s
gather_facts ------------------------------------------------------------ 1.74s
git --------------------------------------------------------------------- 1.26s
etc_hosts --------------------------------------------------------------- 0.04s
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
total ------------------------------------------------------------------ 27.17s
```
